plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt

```diff
@@ -27,9 +40,43 @@ class AmazonQStartupActivity : ProjectActivity {
                 CodeWhispererExplorerActionManager.getInstance().setIsFirstRestartAfterQInstall(false)
             }
         }
-
+        startLsp(project)
         if (runOnce.get()) return
         emitUserState(project)
         runOnce.set(true)
     }
+
+    private fun startLsp(project: Project) {
+        // Automatically start the project context LSP after some delay when average CPU load is below 30%.
+        // The CPU load requirement is to avoid competing with native JetBrains indexing and other CPU expensive OS processes
+        // In the future we will decouple LSP start and indexing start to let LSP perform other tasks.
+        if (CodeWhispererSettings.getInstance().isProjectContextEnabled()) {
+            val scope = CoroutineScope(SupervisorJob())
+            val startLspIndexingDuration = Duration.ofMinutes(30)
+            scope.launch {
+                project.waitForSmartMode()
+                try {
+                    withTimeout(startLspIndexingDuration) {
+                        while (true) {
+                            val cpuUsage = ManagementFactory.getOperatingSystemMXBean().systemLoadAverage
+                            if (cpuUsage > 0 && cpuUsage < 30) {
+                                ProjectContextController.getInstance(project = project)
+                                break
+                            } else {
+                                delay(60_000) // Wait for 60 seconds
+                            }
+                        }
+                    }
+                } catch (e: TimeoutCancellationException) {
+                    LOG.warn(e) { "Failed to start LSP server due to time out" }
+                } catch (e: Error) {
```

Comment:
```suggestion
                } catch (e: Exception) {
```
generally `Error` is considered non-recoverable, so don't catch unless you have a good reason